src/main/java/emu/grasscutter/data/binout/ConfigGadgetCombatProperty.java

```diff
@@ -0,0 +1,41 @@
+package emu.grasscutter.data.binout;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.JsonAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.experimental.FieldDefaults;
+
+import java.io.IOException;
+
+@Data
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class ConfigGadgetCombatProperty {
+    @SerializedName(value="HP", alternate={"CMBBAPKBDCI"})
```

Comment:
I suspect `HP` and `CMBBAPKBDCI` are different fields and `HP` having float and boolean values is a problem with with the resources rip.
<img width="216" alt="image" src="https://user-images.githubusercontent.com/5397662/189954534-53749cf2-ba89-4866-b5ce-714d5e36934e.png">

A quick glance at my resources shows that no file contains both floats and bools for `HP`, and while I haven't checked all of the filenames against dim's repo, it appears that the float `HP` files are from 2.6.0 and the bool `HP` files are from 2.7.0, so I'd call it a wrong name translation on the 2.7.0 binoutput.

I think we'd be better off fixing resources than trying to work around this with gson adapters, especially since `"HP": true` also occurs on gadgets which aren't invulnerable.
<img width="346" alt="image" src="https://user-images.githubusercontent.com/5397662/189958893-bda6cbe0-9d5d-4aee-9a1a-553432f21299.png">

Reply:
dim...

Reply:
Noticed later too, that there seems to be something wrong with the resources themself, after seeing some elements with both CMBBAPKBDCI  and HP.
I updated the branch to remove the TypeAdapter but kept CMBBAPKBDCI as alternate for HP for now.